concurrency/smallTest/ConcurrentLinkedQueueAVOS02.java

```diff
@@ -0,0 +1,96 @@
+package concurrency.smallTest;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+/**
+ * This is for testing MESSAGING ONLY
+ * */
+public class ConcurrentLinkedQueueAtomViolOrOverSync01 {
+    private ConcurrentLinkedQueue<String> queue = new ConcurrentLinkedQueue<>();
+
+    String key = "lala" + 33;
+    String val = "lala" + 44;
+
+    public String give() {
+        queue.remove("key"); // this remove ensures the noRemove() policy does not kick in
+        return queue.peek();
+    }
+
+    
+    //
+    //             !!!!!!!!!!!!                         TESTS MESSAGING ONLY !!!!!!!!!!!!!!!!!!!!!!!
+    //
+    
+    // =========== check size and peek ===========
+
+    public void atomViol04() {
+        String myVal = queue.peek();
+        if (myVal != null) {
+            myVal = queue.peek();
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.



**Problem**  
 You are using a `ConcurrentLinkedQueue`, but your usage of `peek()`, `null` check, and `peek()` may not be thread-safe at lines: **27, 28, and 29**. If the `ConcurrentLinkedQueue` contains only **one element** and another thread removes that element, this thread&#39;s call to `peek()` will get `null`. This `null` may be used at line: **30**. The removes that can remove the element are at lines: **92**, **77**, **62**, **15**, **69**, and **84**.

**Fix**  
 Consider calling `peek()`, checking instead of your current check if the returned object is `null`, and only then using that object.